hydrocarbon_problem/env/env.py

```diff
@@ -21,22 +25,34 @@
 
 class AspenDistillation(dm_env.Environment):
     """
-    Action space assumes unbounded continuous values. This environment has a very not-standard
-    step function (2 streams created from 1 stream - # TODO describe).
+    This environment has a very not-standard step function (2 streams created from 1 stream)
+    To manage this we have an observation object that contains "created states" which are the states
+    that have been recently produced by an action, and "upcoming state" which is the observation of
+    the next stream that will be used to create a distillation column.
+    We also split "done" and "discount" into values for the recently created streams
+    from a specific action - for indicating whether these streams are product streams that leave the process,
+    as well as an "overall" value which indicates whether the overall episode is complete
+    (this occurs when the maximum number of columns have been used, or if there are no non-product
+    streams yet to be acted on in the `._stream_numbers_yet_to_be_acted_on` object.
```

Comment:
Upcoming states are observed after the action has been performed? Read upcoming state is s(t+1), but there is no "expectation" of ending in this certain state, because the action is already performed. (I hope this does not sound vague) 

Do I understand it correctly:
Done: a stream is at specification and "ready to be sold"
Discount: a stream is not at specification?

Reply:
In normal RL you have a chain of states so the upcoming state, and the state produced by an action are the same. We have a weird situation where the upcoming state is the next state in the stream table that can have a column added to it, while 2 other states are produced by a specific action (which I refer to as "created states"). Not sure if this answers your question - but is a bit confusing so we could chat about this in the next meeting. 

Done: correct
Discount: The factor we use for V(s) = r + discount_tops *  value(tops) + discount_bottoms * value(bottoms) 
If the tops or bottoms are product, then we add their `stream_value` to the immediate reward (r), so we set the discount = 0 (as we don't want to also add the value-function estimate for these streams, as the reward has already accounted for their worth and they exit the process so are not acted on further). For streams that aren't products, we set the discount to 1 to account for the "value" of the streams that may be produced further downstream in the process.